android-agent/src/main/kotlin/io/opentelemetry/android/agent/OpenTelemetryRumInitializer.kt

```diff
@@ -130,62 +110,103 @@ object OpenTelemetryRumInitializer {
         return SessionManager.create(timeoutHandler, sessionConfig)
     }
 
-    private fun configureInstrumentation(
-        activityTracerCustomizer: ((Tracer) -> Tracer)?,
-        activityNameExtractor: ScreenNameExtractor?,
-        fragmentTracerCustomizer: ((Tracer) -> Tracer)?,
-        fragmentNameExtractor: ScreenNameExtractor?,
-        anrAttributesExtractors: List<EventAttributesExtractor<Array<StackTraceElement>>>,
-        crashAttributesExtractors: List<EventAttributesExtractor<CrashDetails>>,
-        networkChangeAttributesExtractors: List<NetworkAttributesExtractor>,
-        slowRenderingDetectionPollInterval: Duration?,
-    ) {
-        val activityLifecycleInstrumentation =
-            getInstrumentation<ActivityLifecycleInstrumentation>()
-        if (activityTracerCustomizer != null) {
-            activityLifecycleInstrumentation?.setTracerCustomizer(activityTracerCustomizer)
+    class InstrumentationConfiguration {
```

Comment:
DslMarker might be worth investigating here: https://kotlinlang.org/api/core/kotlin-stdlib/kotlin/-dsl-marker/

Right now it's possible to nest the DSL in a way that doesn't make sense e.g.:

```
instrumentations = {
      activity {
          anrReporter {
              fragment {

              }
          }
      }
  }
```

DslMarker could make that fail at compile-time & therefore help IDE discoverability.